### PR TITLE
synchronize access to Console (fixes #48)

### LINKED ISSUE #48: Windows colored output is mixed up when running in cmd.exe
Windows 10 (version 2004)
go version go1.14.2 windows/amd64

Steps to reproduce:

1. Create "mixed-output.go"
```
package main

import (
    "fmt"
    "github.com/mattn/go-colorable"
    "sync"
)

func main() {
    stdout := colorable.NewColorableStdout()
    wg := &sync.WaitGroup{}
    for i := 0; i < 3; i++ {
       wg.Add(1)
       go func() {
           _, _ = fmt.Fprintln(stdout, "\u001b[32mHello\u001b[0m", "World")
           wg.Done()
       }()
    }
    wg.Wait()
}
```

2. Open Command Prompt (cmd.exe) and cd to file's parent folder.
3. `go build mixed-output.go`
4. Run `mixed-output.exe`, the output is mixed.

Mixed output:
image

Works fine in Git bash (MinGW):
image

### PR DESCRIPTION
**Pull request by segrey**

### CODE REVIEW

**Comment by Fenny** [comment]
@mattn willing to take a quick look? Seems to be a valid PR 👍

File: colorable_windows.go [hunk 1]
```diff
@@ -430,8 +431,14 @@ func atoiWithDefault(s string, def int) (int, error) {
 	return strconv.Atoi(s)
 }
 
+var (
+	writeMutex sync.Mutex
```
Comment (mattn) [inline colorable_windows.go:435]: This mutex have to be a field of Writer.

Comment (segrey) [reply colorable_windows.go]: Thanks, fixed

**Comment by mattn** [comment]
Thank you

**Pull request merged**